### PR TITLE
More unit tests expressionparser

### PR DESCRIPTION
**Pull request by d7919**
Increase expression parser unit test coverage

### CODE REVIEW

File: tests/unit/sys/test_expressionparser.cxx [hunk 1]
```diff
@@ -213,20 +213,25 @@ TEST_F(ExpressionParserTest, BadNumbers) {
...
 }
 
 TEST_F(ExpressionParserTest, BadExpressions) {
+  parser.addGenerator("increment", std::make_shared<IncrementGenerator>());
+
   EXPECT_THROW(parser.parseString("x = 1"), ParseException);
-  EXPECT_THROW(parser.parseString("sin(x"), ParseException);
-  EXPECT_THROW(parser.parseString("sin[x"), ParseException);
+  EXPECT_THROW(parser.parseString("increment(x"), ParseException);
+  EXPECT_THROW(parser.parseString("increment"), ParseException);
```
Comment (d7919) [inline tests/unit/sys/test_expressionparser.cxx]: This has also lost the `[` as well as changing `sin`-->`increment`, is that intentional?

File: tests/unit/sys/test_expressionparser.cxx [hunk 2]
```diff
@@ -213,20 +213,25 @@ TEST_F(ExpressionParserTest, BadNumbers) {
...
   EXPECT_THROW(parser.parseString("2]"), ParseException);
   EXPECT_THROW(parser.parseString("4+"), ParseException);
   EXPECT_THROW(parser.parseString("+4"), ParseException);
   EXPECT_THROW(parser.parseString("\n"), ParseException);
   EXPECT_THROW(parser.parseString("(3"), ParseException);
+  EXPECT_THROW(parser.parseString("2-3[4"), ParseException);
```
Comment (d7919) [inline tests/unit/sys/test_expressionparser.cxx]: Maybe this replaces the above `[` that was removed?

File: tests/unit/sys/test_expressionparser.cxx (hunk 1 above)
Comment (ZedThree) [reply tests/unit/sys/test_expressionparser.cxx]: Yes, the `[` was extraneous for hitting that exception (i.e. it was already throwing before parsing the `[`)

File: tests/unit/sys/test_expressionparser.cxx (hunk 2 above)
Comment (ZedThree) [reply tests/unit/sys/test_expressionparser.cxx]: Not quite, but it does hit a different error [here](https://github.com/boutproject/BOUT-dev/blob/next/src/sys/expressionparser.cxx#L271)

**Pull request merged**